src/main/java/mate/academy/MergeSortAction.java

```diff
@@ -1,8 +1,43 @@
 package mate.academy;
 
-public class MergeSortAction {
+import java.util.Arrays;
+import java.util.concurrent.RecursiveAction;
+
+public class MergeSortAction extends RecursiveAction {
+    private int[] array;
+    private int n;
 
     public MergeSortAction(int[] array) {
+        this.array = array;
+        this.n = array.length;
+    }
+
+    @Override
+    protected void compute() {
+        if (array.length == 1) {
+            return;
+        }
+        var left = Arrays.copyOfRange(array, 0 , n/2);
+        var right = Arrays.copyOfRange(array, n/2 , n);
+        var leftTask = new MergeSortAction(left);
+        var rightTask = new MergeSortAction(right);
+        leftTask.fork();
+        rightTask.compute();
+        leftTask.join();
+        merge(array, left, right);
+    }
 
+    private void merge(int[] array, int[] left, int[] right) {
+        int i = 0;
+        int j = 0;
+        while (i < left.length && j < right.length) {
```

Comment:
The merge loop is correctly merging the two halves, but it's important to ensure that the indices used to access the 'array' are correct throughout the process.